dsi/plugins/tests/test_env.py

```diff
@@ -4,14 +4,24 @@
 
 def test_hostname_plugin_type():
     a = HostnamePlugin()
-    a.parse()
     a.add_row()
     a.add_row()
     assert type(a.output_collector)==collections.OrderedDict
 
 def test_hostname_plugin_col_shape():
     a = HostnamePlugin()
-    a.parse()
     a.add_row()
     a.add_row()
     assert len(a.output_collector.keys())==len(a.output_collector.values())
+
+def test_hostname_plugin_row_shape():
```

Comment:
Would it be better to use a data structure that enforces structure (like a wrapper around OrderedDict that only accepts the original keys given and guarantees an equal number of entries per column, I'm sure there are other options as well) instead of leaving it to the testing phase?

Reply:
I'm not sure. It might be better to engineer a more specific set of requirements for Plugin initialization. My intention here was to provide general examples with specific tests, so the generality supports a Plugin developers creativity, but if/when tests are re-used we get some information that the new Plugin implementation might require different support in the middleware.

I could be convinced one way or the other here. We could:
1. Improve documentation to explain this design goal more clearly.
2. Just do what you say, which might make more sense. I'm torn at the moment...

Reply:
My thought was that as long as we're thinking about structured data, it would be helpful for plugin authors to have a data structure that does not allow malformed output in the first place. This could also be only one option of output format, not mandatory but potentially helpful to still allow that creativity and generality you mention.

Reply:
Like a `strict_enforce` option? I like that. I'll make the change

Reply:
See what you think about this small change. If you can think of a better way, or if I misinterpreted your meaning please commit to this branch with your proposal. Otherwise hit resolve. Thanks!